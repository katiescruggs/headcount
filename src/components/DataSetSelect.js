import React from 'react';
import '../styles/ControlButtons.css';

const DataSetSelect = ({optionNames, changeDataSet, currentDataFile}) => {
  const selectOptions = optionNames.map( (fileName) => {
    return (
      <option key={fileName} value={fileName}>
        {fileName} 
      </option>
    );
  });
  
  return (
    <div className="control-button-container">
      <select
        value={currentDataFile}
        onChange={(e) => changeDataSet(e.target.value)}>
        {selectOptions}
      </select>
    </div>
  );
};

export default DataSetSelect;