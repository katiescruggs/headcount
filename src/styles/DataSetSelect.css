.select-data {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.select-data-instructions {
  margin-top: 0;
}

.select-data select {
  height: 40px;
  font-size: 20px;
}