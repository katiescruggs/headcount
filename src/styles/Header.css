h1 {
  -webkit-margin-before: 0px;
  -webkit-margin-after: 0px;
  -webkit-margin-start: 0px;
  -webkit-margin-end: 0px;
}

.main-hed {
  align-items: center;
  background: #0b1d35;
  color: #ffffff;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  justify-content: space-between;
  padding: 20px 40px 20px 20px;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
}
 
.main-hed h1 {
  font-size: 50px;
}

.logo {
  display: flex;
  flex-direction: row;
}

.top-icon{
  position: relative;
  bottom: 10px;
  left: 10px;
}

@media screen and (max-width: 700px) {
  .main-hed {
    flex-direction: column;
  }
}