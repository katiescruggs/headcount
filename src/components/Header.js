import React from 'react';
import DataSetSelect from './DataSetSelect.js';
import ChildIcon from 'react-icons/lib/fa/child';
import PropTypes from 'prop-types';
import '../styles/Header.css';

const Header = ({dataFileNames, changeDataSet, currentDataFile}) => {
  return (
    <div className="main-hed">
      <div className="logo">
        <h1>
          <span className="top-icon"><ChildIcon /></span>
          Headcount
        </h1>
      </div>


      <DataSetSelect 
        optionNames={dataFileNames}
        changeDataSet={changeDataSet}
        currentDataFile={currentDataFile}/>
    </div>
  );
};

export default Header;

Header.propTypes = {
  dataFileNames: PropTypes.array,
  changeDataSet: PropTypes.func,
  currentDataFile: PropTypes.string
};