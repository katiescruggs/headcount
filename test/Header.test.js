import React from 'react';
import DataSetSelect from '../src/components/DataSetSelect.js';
import Header from '../src/components/Header.js'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'

describe ('Header', () => {
  let header;
  let expectedState ={
    dataFileNames: ['Full Day Kindergartners', 'High School Grad Rates', 'Student Enrollment'], 
    currentDataFile: 'Student Enrollment' 
  };
  const mockFunc = jest.fn()

  beforeEach(() => {
    header = mount(<Header
      dataFileNames={expectedState.dataFileNames}
      CurrentDataFile={expectedState.currentDataFile}
      changeDataSet = {mockFunc} />)

  });

  it('should match the snapshot', () =>{
    expect(toJson(header)).toMatchSnapshot()
  });

  it ('has one select element', () => {
    expect(header.find('select').length).toEqual(1);
  });

  it('runs changeDataSet on change', () =>{
    header.find('select').simulate('change', {target: {value: 'High School Grad Rates'}})
    expect(mockFunc.mock.calls.length).toEqual(1);
  })


})
