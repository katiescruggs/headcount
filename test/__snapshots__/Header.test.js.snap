exports[`Header should match the snapshot 1`] = `
<Header
  CurrentDataFile="Student Enrollment"
  changeDataSet={[Function]}
  dataFileNames={
    Array [
      "Full Day Kindergartners",
      "High School Grad Rates",
      "Student Enrollment",
    ]
  }>
  <div
    className="main-hed">
    <div
      className="logo">
      <h1>
        <span
          className="top-icon">
          <FaChild>
            <IconBase
              viewBox="0 0 40 40">
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em">
                <g>
                  <path
                    d="m32 12.2l-6.5 6.6v18.3q0 1.1-0.7 1.8t-1.8 0.7-1.8-0.7-0.7-1.8v-8.5h-1.4v8.5q0 1.1-0.8 1.8t-1.7 0.7-1.8-0.7-0.7-1.8v-18.3l-6.5-6.6q-0.7-0.6-0.7-1.5t0.7-1.5 1.5-0.6 1.5 0.6l5.1 5.1h8.2l5.1-5.1q0.6-0.6 1.5-0.6t1.5 0.6 0.6 1.5-0.6 1.5z m-7.2-3.6q0 2-1.5 3.5t-3.5 1.5-3.6-1.5-1.4-3.5 1.4-3.6 3.6-1.4 3.5 1.4 1.5 3.6z" />
                </g>
              </svg>
            </IconBase>
          </FaChild>
        </span>
        Headcount
      </h1>
    </div>
    <DataSetSelect
      changeDataSet={[Function]}
      optionNames={
        Array [
          "Full Day Kindergartners",
          "High School Grad Rates",
          "Student Enrollment",
        ]
      }>
      <div
        className="select-data">
        <p
          className="select-data-instructions">
          Select a data set
        </p>
        <select
          onChange={[Function]}>
          <option
            key="Full Day Kindergartners"
            value="Full Day Kindergartners">
            Full Day Kindergartners
          </option>
          <option
            key="High School Grad Rates"
            value="High School Grad Rates">
            High School Grad Rates
          </option>
          <option
            key="Student Enrollment"
            value="Student Enrollment">
            Student Enrollment
          </option>
        </select>
      </div>
    </DataSetSelect>
  </div>
</Header>
`;
